db_manager/test/test_dumper.py

```diff
@@ -0,0 +1,59 @@
+from unittest import TestCase
+from db_manager.dumper import Dumper
+import psycopg2
+import psycopg2.extras
+import os
+import shutil
+
+class TestDumper(TestCase):
+    """Test the class Dumper.
+
+    2 pg_services needed for test related to empty db:
+        qwat_test_1
+        qwat_test_2
+    """
+
+    def setUp(self):
+        self.pg_service1 = 'qwat_test_1'
+        self.pg_service2 = 'qwat_test_2'
+
+        self.conn1 = psycopg2.connect("service={0}".format(self.pg_service1))
+        self.cur1 = self.conn1.cursor()
+
+        self.conn2 = psycopg2.connect("service={0}".format(self.pg_service2))
+        self.cur2 = self.conn2.cursor()
+
+        self.cur1.execute("""
+            DROP SCHEMA IF EXISTS test_dumper CASCADE;
+            CREATE SCHEMA test_dumper;
+            CREATE TABLE test_dumper.dumper_table
+                (
+                id serial NOT NULL,
+                version character varying(50),
+                description character varying(200) NOT NULL,
+                type integer NOT NULL
+                );
+            """)
+        self.conn1.commit()
+
+        self.cur2.execute("""
+            DROP SCHEMA IF EXISTS test_dumper CASCADE;""")
+        self.conn2.commit()
+
+        try:
+            shutil.rmtree('/tmp/test_dumper')
+        except:
+            pass
+
+        os.mkdir('/tmp/test_dumper/')
+
+    def test_dump_restore(self):
+        dumper = Dumper(self.pg_service1, '/tmp/test_dumper/dump.sql')
+        dumper.pg_backup()
+
+        dumper = Dumper(self.pg_service2, '/tmp/test_dumper/dump.sql')
+        dumper.pg_restore()
+
+        # postgres > 9.4
```

Comment:
is postgres strictly > 9.4 or >= 9.4 for this?

Reply:
It should be >=9.4 see [here](https://www.postgresql.org/docs/9.6/static/release-9-4.html). 
I used this only in the unit test modules. In all other places, the sql code is >=9.3 as defined in qwat requirements.

Reply:
It should be >=9.4 see [here](https://www.postgresql.org/docs/9.6/static/release-9-4.html).
I used this only in the unit test modules. In all other places, the sql code is >=9.3 as defined in qwat requirements.